pom.xml

```diff
@@ -2782,6 +2782,13 @@
                     </excludes>
                     <searchTransitive>true</searchTransitive>
                   </bannedDependencies>
+                  <enforceBytecodeVersion>
+                    <maxJdkVersion>${java.version}</maxJdkVersion>
+                    <ignoredScopes>test</ignoredScopes>
+                    <excludes>
```

Comment:
with this change, there still compile failed as follows:

```
[INFO] Spark Project Connect Client ....................... FAILURE [  0.839 s]
[INFO] Spark Project Kinesis Assembly ..................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  22:18 min
[INFO] Finished at: 2023-06-12T14:28:19+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.3.0:enforce (enforce-versions) on project spark-connect-client-jvm_2.12: 
[ERROR] Rule 3: org.codehaus.mojo.extraenforcer.dependencies.EnforceBytecodeVersion failed with message:
[ERROR] Found Banned Dependency: io.get-coursier:interface:jar:1.0.11
[ERROR] Use 'mvn dependency:tree' to locate the source of the banned dependencies.
```

Reply:
```
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- dependency:3.5.0:tree (default-cli) @ spark-connect-client-jvm_2.12 ---
[INFO] org.apache.spark:spark-connect-client-jvm_2.12:jar:3.5.0-SNAPSHOT
[INFO] \- com.lihaoyi:ammonite_2.12.17:jar:2.5.8:provided
[INFO]    \- com.lihaoyi:ammonite-interp-api_2.12.17:jar:2.5.8:provided
[INFO]       \- io.get-coursier:interface:jar:1.0.11:provided
```
`io.get-coursier:interface:jar` is used in provided scope.

Solved this problem by skipping `provided` scope for enforceBytecodeVersion rule.